PyInstaller/building/utils.py

```diff
@@ -509,6 +509,13 @@ def format_binaries_and_datas(binaries_or_datas, workingdir=None):
     toc_datas = set()
 
     for src_root_path_or_glob, trg_root_dir in binaries_or_datas:
+        # Disallow empty source path. Those are typically result of errors, and result in implicit collection of the
+        # whole current working directory, which is never a good idea.
+        if not src_root_path_or_glob:
+            raise SystemExit(
+                "Empty SRC is not allowed when adding binary and data files, as it would result in collection of the "
+                "whole current working directory."
+            )
```

Comment:
If this was hit, the user would have no idea where the empty source path was introduced.

Reply:
Not sure what we can do about that though. We could stick the `dest` path in the message which may tell a PyInstaller savvy user which hook to go rummaging through.

Reply:
Perhaps this check can go when we pull `binaries`/`datas` out of each hook's globals then the error message can point at the offending hook?

Reply:
With the default log level, the preceding line would identify the problematic hook, e.g.:
```
13685 INFO: Loading module hook 'hook-pythoncom.py' from '[...]\\Lib\\site-packages\\_pyinstaller_hooks_contrib\\hooks\\stdhooks'...
Empty SRC is not allowed when adding binary and data files, as it would result in collection of the whole current working directory.
```